src/test/java/com/sonyericsson/rebuild/RebuildValidatorTest.java

```diff
@@ -54,507 +60,524 @@
+     * available on the project level.
+     *
+     * @throws Exception
+     *             Exception
+     */
+    @Test
+    public void testWhenProjectWithoutParamsThenRebuildProjectAvailable()
+            throws Exception {
+        FreeStyleProject project = j.createFreeStyleProject();
+
+        FreeStyleBuild build = project.scheduleBuild2(0).get();
+
+        RebuildLastCompletedBuildAction action = build.getProject().getAction(
+                RebuildLastCompletedBuildAction.class);
+        assertNotNull(action);
+    }
+
+    /**
+     * Creates a new freestyle project and builds the project with a string
+     * parameter. If the build is succesful, a rebuild of the last build is
+     * done. The rebuild on the project level should point to the last build
+     *
+     * @throws Exception
+     *             Exception
+     */
+    @Test
+    public void testWhenProjectWithNoParamsDefinedThenRebuildofBuildWithParamsShouldShowParams()
+            throws Exception {
+        System.setProperty("hudson.model.ParametersAction.keepUndefinedParameters", "true");
+        FreeStyleProject project = j.createFreeStyleProject();
+
+        // Build (#1)
+        project.scheduleBuild2(0, new Cause.UserIdCause(),
+                new ParametersAction(new StringParameterValue("name", "ABC")))
+                .get();
+        HtmlPage rebuildConfigPage = j.createWebClient().getPage(project,
+                "1/rebuild");
+        WebAssert.assertElementPresentByXPath(rebuildConfigPage,
+                "//div[@name='parameter']/input[@value='ABC']");
+    }
+
+    /**
+     * Creates a new freestyle project and builds the project with a string
+     * parameter. If the build is succesful, a rebuild of the last build is
+     * done. The rebuild on the project level should point to the last build
+     *
+     * @throws Exception
+     *             Exception
+     */
+    @Test
+    public void testWhenProjectWithParamsThenRebuildProjectExecutesRebuildOfLastBuild()
+            throws Exception {
+        FreeStyleProject project = j.createFreeStyleProject();
+        project.addProperty(new ParametersDefinitionProperty(
+                new StringParameterDefinition("name", "defaultValue")));
+
+        // Build (#1)
+        project.scheduleBuild2(0, new Cause.UserIdCause(),
+                new ParametersAction(new StringParameterValue("name", "test")))
+                .get();
+        HtmlPage rebuildConfigPage = j.createWebClient().getPage(project,
+                "1/rebuild");
+        // Rebuild (#2)
+        j.submit(rebuildConfigPage.getFormByName("config"));
+
+        HtmlPage projectPage = j.createWebClient().getPage(project);
+        WebAssert.assertLinkPresentWithText(projectPage, "Rebuild Last");
+
+        HtmlAnchor rebuildHref = projectPage.getAnchorByText("Rebuild Last");
+        assertEquals("Rebuild Last should point to the second build", "/jenkins/"
```

Comment:
The only significant change was i had to add `/jenkins/` to get the tests to pass